retrofit-converters/scalars/src/main/java/retrofit2/converter/scalars/ScalarRequestBodyConverter.java

```diff
@@ -23,11 +23,15 @@
 final class ScalarRequestBodyConverter<T> implements Converter<T, RequestBody> {
   static final ScalarRequestBodyConverter<Object> INSTANCE = new ScalarRequestBodyConverter<>();
   private static final MediaType MEDIA_TYPE = MediaType.parse("text/plain; charset=UTF-8");
+  private static final RequestBody EMPTY_BODY = RequestBody.create(null, new byte[0]);
 
   private ScalarRequestBodyConverter() {
   }
 
   @Override public RequestBody convert(T value) throws IOException {
+    if (value == null) {
```

Comment:
I think this behavior is pretty strange. Is it essential?

The problem is that `""` is a valid string, but not a valid integer.